Automattic/jetpack
# Migrate v4 IDC endpoints into package

Migrate the existing IDC endpoints into the IDC package. This includes removing dependencies on Jetpack Disconnect.

#### Changes proposed in this Pull Request:

* Packaging of Identity Crisis Functionality.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


NO

#### Testing instructions:






Regression Testing

Verify that new site registration have home_url and site_url as expected on the WP.com Cache (E2E flows cover)

Verify that disconnecting Jetpack
-- remove site connection / token
-- remove user connections / tokens
-- updates activated Jetpack_Option to 4
-- updates unique_connection Jetpack_Option's disconnected value by 1

Verify that Identity Crisis flows are fully functional 
-- Duplicate your JP connected site to a new site (update options to new url except leave jetpack_private_options untouched)
-- Perform an edit to a post (triggers sync)
-- Visit the Jetpack Dashboard
-- Safe Mode will be enabled and IDC options presented.
-- Verify that choosing this is a new site, doesn't modify the home_url or site_url of the existing cache site and instead the new site has its own cache site created and jetpack_private_options updated.
-- Verify that choosing this is the site has stats migrated and home_url and site_url of the existing cache site is updated.
Modify a site's site_url and home_url. Ensure these are synced to the WP.com Cache (unit tests cover)

Note #1: Quickest way to trigger IDC flows is to copy jetpack_private_options from Site A to Site B then trigger a content change on Site B (edit post, etc)

Note #2 :: Full step by step instructions in How to Test various IDC / Safe Mode functionality is outlined in the "How to Test" section of https://wp.me/p1HpG7-3EC
